### PR TITLE
[Add-ins][Admin] Fixes bug 3720906

### PR DESCRIPTION
**Pull request by mscharlock**
This article used to link to a really old stack - I think instead of using setTimeout, now you'd use async but thought I'd leave it there.

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[42b615f](https://github.com/OfficeDev/office-js-docs-pr/commits/42b615f7fa39ea6782070657e8ec21f189b23502)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/concepts/resource-limits-and-performance-optimization.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-settimeout/docs/concepts/resource-limits-and-performance-optimization.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/concepts/resource-limits-and-performance-optimization?branch=pr-en-us-2015) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C7%5C13%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202007132127570268-2015%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=qoDsjncXsE7HCJvRhoFB1g5KW8OlZVphx1us2PJwzoo%3D&st=2020-07-13T21%3A24%3A38Z&se=2020-08-13T21%3A29%3A38Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

File: docs/concepts/resource-limits-and-performance-optimization.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ While the resources limits on CPU and memory usage, crash tolerance, UI responsi
...
 
-   For a JavaScript and jQuery code sample that shows breaking up a potentially long-running and CPU-intensive series of inputting and outputting operations on unbounded data, see [How can I give control back (briefly) to the browser during intensive JavaScript processing?](https://stackoverflow.com/questions/210821/how-can-i-give-control-back-briefly-to-the-browser-during-intensive-javascript). This example uses the [setTimeout](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) method of the global object to limit the duration of input and output. It also handles the data in defined chunks instead of randomly unbounded data.
+You can do this through the [setTimeout](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) method of the global object to limit the duration of input and output. It also handles the data in defined chunks instead of randomly unbounded data. Another option is to use [async](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) to handle your Promises.
```
Comment (ElizabethSamuel-MSFT) [inline docs/concepts/resource-limits-and-performance-optimization.md:94]: Is this indented properly? Shouldn't it fall under the first bullet?

File: docs/concepts/resource-limits-and-performance-optimization.md [hunk 2]
```diff
@@ -91,7 +91,7 @@ While the resources limits on CPU and memory usage, crash tolerance, UI responsi
...
+You can do this through the [setTimeout](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) method of the global object to limit the duration of input and output. It also handles the data in defined chunks instead of randomly unbounded data. Another option is to use [async](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) to handle your Promises. 
 
 - If your add-in uses a CPU-intensive algorithm to process a large volume of data, you can use web workers to perform the long-running task in the background while running a separate script in the foreground, such as displaying progress in the user interface. Web workers do not block user activities and allow the HTML page to remain responsive. For an example of web workers, see [The Basics of Web Workers](https://www.html5rocks.com/en/tutorials/workers/basics/). See [Web Workers](https://developer.mozilla.org/docs/Web/API/Web_Workers_API) for more information about the Web Workers API.
```
Comment (ElizabethSamuel-MSFT) [inline docs/concepts/resource-limits-and-performance-optimization.md:96]: Can you make the web workers link locale agnostic?

File: docs/concepts/resource-limits-and-performance-optimization.md (hunk 1 above)
Comment (ElizabethSamuel-MSFT) [reply docs/concepts/resource-limits-and-performance-optimization.md:94]: Can you use a locale-agnostic link for the async function?

**Comment by o365devx** [comment]
Docs Build status updates of commit _[548d27c](https://github.com/OfficeDev/office-js-docs-pr/commits/548d27c88d0e973b974c4e2f58f69d7cb9d8e9ff)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/concepts/resource-limits-and-performance-optimization.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-settimeout/docs/concepts/resource-limits-and-performance-optimization.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/concepts/resource-limits-and-performance-optimization?branch=pr-en-us-2015) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C7%5C13%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202007132150376435-2015%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=Kxb1Bdslto0eOiZLO%2F3TxuvrlXu7Cjlbi0NgK3pVsoM%3D&st=2020-07-13T21%3A46%3A55Z&se=2020-08-13T21%3A51%3A55Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Comment by mscharlock** [comment]
@Rick-Kirkham I am pretty sure we can get rid of the link to this old Stack Overflow, but I want to make sure that what I've written here still holds up. Just wanted you to take a quick look at it when you get a sec. Thanks!

**Comment by lindalu-MSFT** [comment]
Is this good to publish?

**Comment by lindalu-MSFT** [comment]
This has been sitting 22 days, going to merge. If there are more updates, please create a new PR! Thanks

**Pull request merged**